cpp/include/cudf/legacy/table.hpp

```diff
@@ -112,7 +112,9 @@ struct table {
    *---------------------------------------------------------------------------**/
   gdf_column* get_column(gdf_index_type index) {
     assert(index < _columns.size());
-    return _columns[index];
+    if(index < gdf_size_type(_columns.size()))
```

Comment:
The `assert` already takes care of bounds checking in Debug builds. In general, I don't think we should be doing bounds checking in Release builds.